server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -1367,4 +1387,35 @@ private Result handleUpdateExceptions(Throwable throwable) {
     }
     throw new RuntimeException(throwable);
   }
+
+  /**
+   * Check if the application needs to be updated to a newer program version. If it does updated and
+   * return a redirect result back to the review page
+   *
+   * @return {@link Result} if application was updated; empty if not
+   */
+  private Optional<Result> updateApplicationToLatestProgramVersionIfNeeded(
```

Comment:
Question (Blocking): This method looks very similar to the one below. Can it live in a common file and be referenced by each caller?

Reply:
They are indeed exactly the same and I expected this to come up as I went back and forth trying to decide on where to put it. :) 

Options include:
- Pushing to the base controller, but that adds more dependencies and will require touching anything that needs to now pass in those new dependencies to the base class. The touch is too broad, especially for just being used in two places.
- Move it to the service class, but now the service classes need to know about routing and URLs. Controller stuff that it shouldn't know.
- Move it to a new class and either call it from a static method or add another thing to the already long list of things being injected. I think it's pretty specialized and creating a new class seems overboard. 
- Leave it as is, duplicated in two controllers. Downside a tiny bit of duplication. Upside maintains a locality to where it's actually used.

I'm inclined to leave it as it. I'm open to suggestions if you like one of the other options or have another idea.

Reply:
I don't have a strong opinion, but I would choose creating a new class with a static method. I'm afraid that someone will eventually update one but not the other.

I'm OK with submitting as-is, so feel free to close this thread